src/Form.php

```diff
@@ -49,6 +49,18 @@ public function addField(string $name, string $content, ?string $contentType = n
         $this->fields[] = new FormField($name, BufferedContent::fromString($content, $contentType));
     }
 
+    /**
+     * Adds each member of the array as an entry for the given key name. Array keys are persevered.
+     *
+     * @param array<string> $fields
+     */
+    public function addFields(string $name, array $fields): void
+    {
+        foreach ($fields as $key => $content) {
+            $this->addField(\sprintf('%s[%s]', $name, $key), $content);
```

Comment:
This is PHP specific syntax, no?

Reply:
No, it seems this is a de-facto "standard" for arrays in query strings. I found several libraries supporting it. Here's one for example in python: https://github.com/bernii/querystring-parser

Reply:
Should it support any depth then?

Reply:
Sure, probably uncommon, but that makes sense. Updated.